docs/proposals/policy.md

```diff
@@ -0,0 +1,146 @@
+  5.  create
+  6.  update
+  7.  delete
+  8.  proxy
+
+
+### Verb Specific Attributes
+Different verbs will require different attributes.  For instance, the exec verb will have information about the command being run, while a get will have the resource kind being requested, and an update may have a list of the fields being modified.  The exact contract rules will be chosen by the Authorizer/AuthorizationAttributeBuilder pair, but basic resource access for openshift should include  kind.
+
+
+### Configtime Authorization Types
+```
+// PolicyRule holds information that describes a policy rule, but does not contain information 
+// about who the rule applies to or which namespace the rule applies to.  This makes PolicyRules portable units.
+type PolicyRule struct {
+	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
+	Deny bool
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule
+	Verbs []Verb
+	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.  
+	// Making it an interface{} allows a type switch for identification followed by properly typed usage thereafter.
+	AttributeRestrictions interface{}
+	// ResourceKinds is a list of kinds this rule applies to
+	ResourceKinds []string
+}
+
+// PolicyRuleBinding builds on top of a PolicyRule.  It adds who information (usernames, teamnames, accountnames) 
+// and namespace information (ObjectMeta).  PolicyRuleBindings only apply to the namespace they are contained in.  
+// That means the full set of PolicyRuleBindings for a namespaced request are contained inside of the requested 
+// namespace or in the global namespace
+type PolicyRuleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	Rule PolicyRule
+}
+
+// Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
+type Role struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []PolicyRule
+}
+
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information like a PolicyRuleBinding.  
+// It can reference any Role in the same namespace or in the global namespace.
+type RoleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	// this is a ref to a Role
+	ReferencesGlobalRole bool
+	RoleName             string
+}
+```
+
+### What is a global namespace?
+The types used for policy configuration need to be namespaced types to enable efficient retrieval during policy evaluation, but some Policies, Roles, and their associated bindings need to be globally accessable as well.  One way to support this is to create a special global namespace called "MotherBrain".  It would be a reserved, well-known name that would allow policy to be written against it.
```

Comment:
> Just saying magic value here makes this whole concept on unstable ground. Why is this namespace special?

The global namespace is special because the concept of policy that applies to every namespace is valuable.  Having policy rule bindings namespaced is important because of the retrieval options available in etcd (no lookups by field).   The default namespace doesn't seem like a logical location.

@derekwaynecarr I don't want to speak for you, but if memory serves you were facing a similar problem in admission control.  Global limits that should be described by objects that are stored in a namespaced fashion.